packages/driver/src/cy/commands/files.ts

```diff
@@ -43,57 +47,71 @@ export default (Commands, Cypress, cy) => {
         })
       }
 
-      const verifyAssertions = () => {
-        return Cypress.backend('read:file', file, _.pick(options, 'encoding'))
-        .catch((err) => {
+      const verifyAssertions = async () => {
+        let result
+
+        try {
+          result = await Cypress.backend('read:file', file, _.pick(options, ['encoding'])).timeout(options.timeout)
+        } catch (err) {
           if (err.code === 'ENOENT') {
-            return {
+            result = {
               contents: null,
               filePath: err.filePath,
             }
+          } else if (err.name === 'TimeoutError') {
+            return $errUtils.throwErrByPath('files.timed_out', {
+              onFail: options._log,
+              args: { cmd: 'readFile', file, timeout: options.timeout },
+            })
+          } else {
+            return $errUtils.throwErrByPath('files.unexpected_error', {
+              onFail: options._log,
+              args: { cmd: 'readFile', action: 'read', file, filePath: err.filePath, error: err.message },
+            })
```

Comment:
nit: there's no reason to use `else` here

Reply:
Without the else, we'll throw in the case of a ENOENT error, which we're trying to avoid. I could restructure this a bit to be more clear?